### PR TITLE
Added parametric test for Flux BC initialise

### PR DESCRIPTION
**Pull request by gabriele-ferrero**
## Proposed changes

New parametric test to test the integration of several boundary conditions.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [X] New tests

## Checklist

- [X] Black formatted
- [X] Unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: test/unit/test_boundary_conditions.py [hunk 1]
```diff
@@ -543,15 +543,23 @@ def test_string_for_field_in_dirichletbc():
     bc.create_dirichletbc(V, fenics.Constant(1), surface_marker)
 
 
-def test_dissociation_flux_solve():
-    """Test to catch bug DissociationFlux see #581"""
+@pytest.mark.parametrize(
+    "boundary",
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:547]:
```suggestion
    "bc",
```

File: test/unit/test_boundary_conditions.py [hunk 2]
```diff
@@ -543,15 +543,23 @@ def test_string_for_field_in_dirichletbc():
...
+    [
+        (festim.DissociationFlux(surfaces=[1], Kd_0=1, E_Kd=0, P=1e4)),
+        (festim.ConvectiveFlux(h_coeff=1, T_ext=1, surfaces=1)),
+        (festim.FluxBC(surfaces=1, value=1, field=0)),
+        (festim.MassFlux(h_coeff=1, c_ext=1, surfaces=1)),
+        (festim.RecombinationFlux(Kr_0=1e-20, E_Kr=0, order=2, surfaces=1)),
+    ],
+)
+def test_flux_BC_initialise(boundary):
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:556]:
```suggestion
def test_flux_BC_initialise(bc):
```

File: test/unit/test_boundary_conditions.py [hunk 3]
```diff
@@ -543,15 +543,23 @@ def test_string_for_field_in_dirichletbc():
...
+    """Test to catch bug Flux BCs see #581"""
     sim = festim.Simulation()
     sim.mesh = festim.MeshFromVertices([0, 1, 2, 3])
     sim.materials = festim.Material(id=1, D_0=1, E_D=0)
     sim.T = festim.Temperature(value=500)
-    sim.boundary_conditions = [
-        festim.DissociationFlux(surfaces=[1], Kd_0=1, E_Kd=0, P=1e4)
-    ]
+    sim.boundary_conditions = [boundary]
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:562]:
```suggestion
    sim.boundary_conditions = [bc]
```

File: test/unit/test_boundary_conditions.py [hunk 4]
```diff
@@ -543,15 +543,23 @@ def test_string_for_field_in_dirichletbc():
...
-def test_dissociation_flux_solve():
-    """Test to catch bug DissociationFlux see #581"""
+@pytest.mark.parametrize(
+    "boundary",
+    [
+        (festim.DissociationFlux(surfaces=[1], Kd_0=1, E_Kd=0, P=1e4)),
+        (festim.ConvectiveFlux(h_coeff=1, T_ext=1, surfaces=1)),
+        (festim.FluxBC(surfaces=1, value=1, field=0)),
+        (festim.MassFlux(h_coeff=1, c_ext=1, surfaces=1)),
+        (festim.RecombinationFlux(Kr_0=1e-20, E_Kr=0, order=2, surfaces=1)),
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:553]: We could be even more thorough here and add some Dirichlet boundary conditions to this list!

**Review by RemDelaporteMathurin — COMMENTED** [review]
This is a super idea @gabriele-ferrero thank you! 🚀 

This is a good example on how to scale up tests.

Very few comments on my side.
We could add more boundary conditions and also test for dirichlet bcs

@jhdark feel free to pitch in

**Review by RemDelaporteMathurin — APPROVED** [review]
LGTM 🚀
